[
  {
    "type": "MCQ",
    "question": "What is the output of the above code?",
    "code": "class Main {\r\n    public static void main(String[] args) {\r\n        System.out.println(print(1));\r\n    }\r\n\r\n    static Exception print(int i) {\r\n        if (i > 0) {\r\n            return new Exception();\r\n        } else {\r\n            throw new RuntimeException();\r\n        }\r\n    }\r\n}",
    "correct_answers": ["`java.lang.Exception`"],
    "incorrect_answers": [
      "A RuntimeException is thrown",
      "A Exception is thrown",
      "`java.lang.RuntimeException`",
      "This code will not compile"
    ],
    "explanation": "There is a difference between throw and return."
  },
  {
    "type": "MRQ",
    "question": "Choose all the problematic ways of handling exceptions above. (Problematic meaning unable to compile OR unnecessary)",
    "code": "void willThrow1() {\r\n    // does some stuff\r\n    throw new NullPointerException(\"1n\");\r\n  }\r\n  void willThrow2() throws NullPointerException {\r\n    throw new NullPointerException(\"2\");\r\n  }\r\n  void willThrow3() throws IOException {\r\n    // does some stuff\r\n    throw new IOException();\r\n  }\r\nvoid willThrow4() {\r\n    throw new IOException();\r\n}\r\n",
    "correct_answers": ["`void willThrow2()`", "`void willThrow4()`"],
    "incorrect_answers": ["`void willThrow1()`", "`void willThrow3()`"],
    "explanation": "How to handle checked vs unchecked exceptions? Type out the code in jshell and experiment with it. If not sure please ask."
  }
]
